OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -296,6 +296,14 @@ public WDist GetMaximumRangeVersusTarget(Target target)
 		// Enumerates all armaments, that this actor possesses, that can be used against Target t
 		public IEnumerable<Armament> ChooseArmamentsForTarget(Target t, bool forceAttack)
 		{
+			// If the target is frozen, make sure it is still valid
+			if (t.Type == TargetType.FrozenActor)
+			{
+				var frozenLayer = self.Owner.PlayerActor.TraitOrDefault<FrozenActorLayer>();
+				if (frozenLayer == null || frozenLayer.FromID(t.FrozenActor.ID) == null)
```

Comment:
@pchote - I believe I fixed this (it seems like there was only one place that needed fixing in the method, the rest either check the type before accessing the actor field, or use `CenterPosition`)